File: Java/가운데 글자 가져오기.java
// 가운데 글자 가져오기
public class Q5 {
    public static void main(String[] args) {
        String s = "abcde";
        String s1 = "qwer";
        MiddlePick middlePick = new MiddlePick();
        System.out.println(middlePick.middle(s));
        System.out.println(s.charAt(s.length()/2 + 1));

    }
}
class MiddlePick{
    public String middle(String s){
        String answer = "";
//        if(s.length()%2 == 0){
//            answer = ""+(s.charAt(s.length()/2-1))+(s.charAt(s.length()/2));
//        }
//        else {
//            answer = s.charAt(s.length() / 2) + "";
//        }

        return s.substring((s.length()-1) / 2, s.length()/2 + 1); // 이렇게 작성하면 코드가 줄어듬
    }
}

File: JavaScript/가운데 글자 가져오기.js
/*
가운데 글자 가져오기 : 
  단어 s의 가운데 글자를 반환하는 함수를 완성하세요.
  단어의 길이가 짝수라면 가운데 두글자를 반환하면 됩니다.
제한 조건 : 
  s의 길이는 1 이상 100 이하인 스트링
  */
const solution = s => s.length%2===0? s.charAt(s.length/2-1)+s.charAt(s.length/2) : s.charAt(s.length/2);

console.log(solution("abcde")); // c
console.log(solution("qwer")); // we